hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -314,7 +314,8 @@ public void put(K key, V value) {
         R oldRecord = null;
         try {
             record = valueToRecord(value);
-            onRecordCreate(key, record);
+            onRecordCreate(record);
+            initInvalidationMetaData(record, key, keyData);
```

Comment:
This is a second code path where the invalidation metadata is set. The change is uncritical, since the metadata is still set on the record creation.